tests/components/minecraft_server/test_init.py

```diff
@@ -108,115 +115,192 @@ def create_v1_mock_binary_sensor_entity_entry(
+            java_mock_config_entry.entry_id
+        )
+
+    await hass.async_block_till_done()
+    assert java_mock_config_entry.state == ConfigEntryState.SETUP_ERROR
+
+
+async def test_setup_entry_not_ready(
+    hass: HomeAssistant, java_mock_config_entry: MockConfigEntry
+) -> None:
+    """Test entry setup not ready."""
+    java_mock_config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.minecraft_server.api.JavaServer.lookup",
+        return_value=JavaServer(host=TEST_HOST, port=TEST_PORT),
+    ), patch(
+        "homeassistant.components.minecraft_server.api.JavaServer.async_status",
+        return_value=OSError,
+    ):
+        assert not await hass.config_entries.async_setup(
+            java_mock_config_entry.entry_id
+        )
+
+    await hass.async_block_till_done()
+    assert java_mock_config_entry.state == ConfigEntryState.SETUP_RETRY
+
+
+async def test_entry_migration(
+    hass: HomeAssistant, v1_mock_config_entry: MockConfigEntry
+) -> None:
     """Test entry migration from version 1 to 3, where host and port is required for the connection to the server."""
-    config_entry_id = create_v1_mock_config_entry(hass)
-    device_entry_id = create_v1_mock_device_entry(hass, config_entry_id)
+    v1_mock_config_entry.add_to_hass(hass)
+
+    device_entry_id = create_v1_mock_device_entry(hass, v1_mock_config_entry.entry_id)
     sensor_entity_id_key_mapping_list = create_v1_mock_sensor_entity_entries(
-        hass, config_entry_id, device_entry_id
+        hass, v1_mock_config_entry.entry_id, device_entry_id
     )
     binary_sensor_entity_id_key_mapping = create_v1_mock_binary_sensor_entity_entry(
-        hass, config_entry_id, device_entry_id
+        hass, v1_mock_config_entry.entry_id, device_entry_id
     )
 
     # Trigger migration.
     with patch(
-        "homeassistant.components.minecraft_server.api.MinecraftServer.__init__",
+        "homeassistant.components.minecraft_server.api.JavaServer.lookup",
         side_effect=[
-            MinecraftServerAddressError,  # async_migrate_entry
-            None,  # async_migrate_entry
-            None,  # async_setup_entry
+            ValueError,  # async_migrate_entry
+            JavaServer(host=TEST_HOST, port=TEST_PORT),  # async_migrate_entry
+            JavaServer(host=TEST_HOST, port=TEST_PORT),  # async_setup_entry
         ],
-        return_value=None,
     ), patch(
-        "homeassistant.components.minecraft_server.api.MinecraftServer.async_get_data",
-        return_value=TEST_JAVA_DATA,
+        "homeassistant.components.minecraft_server.api.JavaServer.async_status",
+        return_value=TEST_JAVA_STATUS_RESPONSE,
     ):
-        assert await hass.config_entries.async_setup(config_entry_id)
+        assert await hass.config_entries.async_setup(v1_mock_config_entry.entry_id)
         await hass.async_block_till_done()
 
+    migrated_config_entry = v1_mock_config_entry
```

Comment:
These are the same object throughout the test so i'm not sure that it's worth renaming it given its the same config entry the whole time.

Reply:
I found it a bit weird to work with `v1_` in the name of the config entry after the migration is done. I just wanted to make it more clear. But if you want I can remove it again?

Reply:
Or we could use a different name for the fixture. A possible solution would be:
```
@pytest.mark.parametrize("mock_config_entry", ("v1_mock_config_entry",))
async def test_entry_migration(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    request: pytest.FixtureRequest,
) -> None:
    """Test entry migration from version 1 to 3, where host and port is required for the connection to the server."""
    mock_config_entry = request.getfixturevalue(mock_config_entry)
    mock_config_entry.add_to_hass(hass)
    [...]
```

Reply:
OK we can leave as is.